### PR TITLE
Start of migration tooling

### PR DESCRIPTION
**Pull request by TimWhiting**
@rrousselGit 
Ran into some writer's block on the documentation, but was interested in trying out codemod's interface.

Here is an initial implementation of the migration tooling using `package:codemod`. One limitation of the package that I found is that it is not null-safe yet. Hopefully it still will be able to use a version of the analyzer that can resolve/consume null-safe dart. I've submitted an issue on their repository for migration to null-safety, and will keep an eye on that.

Would appreciate some feedback on this.

### CODE REVIEW

**Comment by rrousselGit** [comment]
Analyzer itself isn't null safe, but it can analyze null safe code.
Code generators all have this "issue". So don't worry about it

**Comment by rrousselGit** [comment]
Would it make sense to have the bin inside `riverpod` instead of a separate `codemod_riverpod`?

File: packages/codemod_riverpod/test/imports_test.dart [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  group('ImportAllRenamer', () {
+    test('renames imports', () {
+      final sourceFile = SourceFile.fromString('''
+// Don't touch path
+import 'package:path/path.dart';
+
+// Changes required
+import 'package:riverpod/all.dart';
+import 'package:hooks_riverpod/all.dart';
+import 'package:flutter_riverpod/all.dart';
```
Comment (rrousselGit) [inline packages/codemod_riverpod/test/imports_test.dart:16]: Good idea!

**Comment by TimWhiting** [comment]
> Would it make sense to have the bin inside `riverpod` instead of a separate `codemod_riverpod`?

I think as there gets to be more migrations, there will be a need for more files in lib rather than putting everything in bin. And since the `codemod_riverpod` package depends on `analyzer` in regular `dependencies` and not `dev_dependencies` that isn't something that we want to impose on the `riverpod` package.

**Comment by TimWhiting** [comment]
If you want I can put everything in the `riverpod` bin though.

**Comment by TimWhiting** [comment]
If you want I can put everything in the `riverpod` bin though.

**Comment by rrousselGit** [comment]
No that's fine, don't worry about it

**Comment by TimWhiting** [comment]
In working on figuring out a migration for #341 I found out that `codemod` does not provide a resolved dart AST, so I couldn't find out the static types for determining what kind of provider they were watching. So I added that capability to the `codemod` package and submitted a [PR](https://github.com/Workiva/dart_codemod/pull/38). Currently using my local changes to work on the migration. 

Also, do you think I should create a separate entrypoint / executable for each migration (i.e. migrate12.3to4 for the import changes, and maybe migrate13to14 for the upcoming breaking changes assuming they come in 14.0 since null safety was the breaking change for 13.0).

**Comment by rrousselGit** [comment]
I see thanks!


We should be able to have a single entry point, and determine what to use based on the version in the pubspec/lock file.

**Comment by rrousselGit** [comment]
Thanks, I'll merge this and continue later

**Pull request merged**